src/ripple/rpc/handlers/PayChanClaim.cpp

```diff
@@ -90,15 +90,31 @@ Json::Value doChannelVerify (RPC::Context& context)
 {
     auto const& params (context.params);
     for (auto const& p :
-        {jss::public_key, jss::channel_id, jss::amount, jss::signature})
+                 {jss::public_key, jss::channel_id, jss::amount, jss::signature})
         if (!params.isMember (p))
             return RPC::missing_field_error (p);
 
-    std::string const strPk = params[jss::public_key].asString ();
-    auto const pk =
-        parseBase58<PublicKey> (TokenType::TOKEN_ACCOUNT_PUBLIC, strPk);
-    if (!pk)
-        return rpcError (rpcPUBLIC_MALFORMED);
+    boost::optional<PublicKey> pk;
+    {
+        std::string const strPk = params[jss::public_key].asString();
+        pk = parseBase58<PublicKey>(TokenType::TOKEN_ACCOUNT_PUBLIC, strPk);
+
+        if (!pk)
+        {
+            std::pair<Blob, bool> pkHex(strUnHex (strPk));
+            if (!pkHex.second)
+                return rpcError(rpcPUBLIC_MALFORMED);
```

Comment:
Suggestion: This code path is not covered by unit test. We can improve the coverage and test.